packages/scandipwa/src/component/CheckoutAddressBook/CheckoutAddressBook.container.js

```diff
@@ -151,10 +165,74 @@ export class CheckoutAddressBookContainer extends PureComponent {
             isBilling,
             selectedAddressId,
             isSubmitted,
-            is_virtual
+            shippingFields
         };
     }
 
+    getAddressIdFromAddressBook() {
+        const {
+            shippingFields,
+            customer: {
+                addresses = []
+            } = {}
+        } = this.props;
+
+        // the main keys distinguishing addresses
+        const keys = [
+            'city',
+            'country_id',
+            'firstname',
+            'lastname',
+            'postcode',
+            'telephone',
+            'vat_id'
+        ];
+
+        const filteredAddresses = (
+            addresses.filter((address) => (
+                keys.every((key) => {
+                    if (shippingFields[key] !== address[key]) {
+                        return false;
+                    }
+
+                    return true;
+                })
+                && (
+                    JSON.stringify(shippingFields.street) === JSON.stringify(address.street)
+                )
+            ))
+        );
+
+        return filteredAddresses.length ? filteredAddresses[0].id : 0;
+    }
```

Comment:
1. This can be as util
2. Isn't it should be for only customer if customer actually logined? 
3. Can we just safe customer address id in shipping fields or somewhere else (redux) to reuse it later for this case? So we won't needed to go through all address.